addons/common/functions/fnc_setupLocalUnitsHandler.sqf

```diff
@@ -0,0 +1,76 @@
+/*
+ * Author: dedmen  
+ * Adds the local unit event handlers. Access with array `ace_common_localUnits`.
+ *
+ * Arguments:
+ * Nothing
+ *
+ * Return Value:
+ * Nothing
+ *
+ * Example:
+ * [] call ace_common_fnc_setupLocalUnitsHandler
+ *
+ * Public: No
+ */
+// #define DEBUG_MODE_FULL
+#include "script_component.hpp"
+
+GVAR(localUnits) = [];
+
+//Eventhandlers to maintain array of localUnits
+["CAManBase", "init", {
+    params ["_unit"];
+    TRACE_2("unit init",_unit,local _unit);
+
+    if (local _unit) then {
+        if (!alive _unit) exitWith {};
+        GVAR(localUnits) pushBack _unit;
+    };
+}] call CBA_fnc_addClassEventHandler;
+
+["CAManBase", "respawn", {
+    params ["_unit"];
+    TRACE_2("unit respawn",_unit,local _unit);
+
+    if (local _unit) then {
+        if (!alive _unit) exitWith {};
+        GVAR(localUnits) pushBack _unit;
+    };
+}] call CBA_fnc_addClassEventHandler;
+
+["CAManBase", "local", {
+    params ["_unit", "_local"];
+    TRACE_2("unit local",_unit,_local);
+
+    if (_local) then {
+        if (!alive _unit) exitWith {};
+        GVAR(localUnits) pushBack _unit;
+    } else {
+        GVAR(localUnits) deleteAt (GVAR(localUnits) find _unit);
+    };
+}] call CBA_fnc_addClassEventHandler;
+
+["CAManBase", "deleted", {
+    params ["_unit"];
+    TRACE_2("unit deleted",_unit,local _unit);
+
+    if (local _unit) then {
+        [{
+            params ["_unit"];
+            TRACE_3("unit deleted nextFrame",_unit,local _unit,isNull _unit);
+            if (isNull _unit) then { //If it is not null then the deleted EH was Fake.
+                GVAR(localUnits) = GVAR(localUnits) - [objNull];
```

Comment:
I delete all objNull here as using `find` to find a null object will find the first null object and not the one the deleted EH belongs to.
I would prefer to just `call FUNC(getLocalUnits)` here in case we add a check to getLocalUnits so it only removes null objects once per frame.
But it would hurt readability.